subsys/dfu/boot/mcuboot.c

```diff
@@ -144,6 +144,11 @@ int boot_read_bank_header(uint8_t area_id,
 	return 0;
 }
 
+int boot_read_img_header(struct mcuboot_img_header *header, size_t header_size)
+{
+	return boot_read_bank_header(FLASH_AREA_IMAGE_PRIMARY, header, header_size);
```

Comment:
This is not correct, this only works whereby slot0 is the currently running image so does not work for direct XIP or RAM load

Reply:
What would be the appropriate way to get the current header for those cases?

Reply:
You have to figure out where is your code running from. There are several ways to approach this. You can, at compile time, store somewhere slot or offset your images runs from.

Reply:
There is code in MCUmgr that does that (I can point you later to that part) or you could possibly use DTS chosen zephyr,code-partition.

Reply:
Determining the offset or flash area ID of the currently running image is not difficult, as you mention.
The challenge is mapping that back to the correct `FLASH_AREA_IMAGE_` define.

It is also not true that the location where your code is running is the same place that you need to look to find the MCUboot header. The primary case for this is when you have secure and non-secure images merged together for the purposes of MCUboot.

Reply:
We are generating DTS function for scanning through all defined partitions.
If we are missing something else to match the boot partition, what else should we generate/define at compile time?